### DISCUSSION TITLE
How can i set specific cluster from KUBECONFIG file

### DISCUSSION THREAD

**Discussion by SomeAkk** [redhat-developer/intellij-kubernetes#555] [Q&A] [answered]
By default i see local k8s cluster (in my case it is docker-desktop cluster by DockerDesktop). I want to to see non-local clusters in sidebar too. 

Can i just put anywhere KUBECONFIG file of remote cluster? 
Maybe i need set some env variable?

Thx.

**Comment by SomeAkk** [comment]
@mohitsuman does discussions work or need to write to issues?

**Comment by mohitsuman** [comment]
Hi @SomeAkk thanks for raising this discussion, I will let @adietish put his views here.

**Comment by adietish** [comment, marked as answer]
Hi @SomeAkk 

We are currently tied to the default location at ~/.kube/config unfortunately. You should be able to work around it with a symlink though. 
I created an issue for this at #586 and set it to get tackled in the current sprint. It should get released this months in binary form.

**Reply by SomeAkk** [reply]
Ok, it is good workaround to use default ~/.kube/config - will use it while await [#586](https://github.com/redhat-developer/intellij-kubernetes/issues/586) resolving.

Thanks for your answer!

For your info: my case grew from that fact, that i need often to change k8s clusters. Now another tools, which i use, can look at KUBECONFIG env. By setting KUBECONFIG to my location of ~/awesome_place/kube/stand_#1/configs/admin.conf globally i got tools working with stand_#1 (and so on with stand_#2, stand_#3, ...).

Maybe it will help you to closer understand why i asked you for that feature.
I thing it will be good for your tool works in such way.

**Reply by adietish** [reply]
oh yes, it does. Thanks! Appreciate your insight.
Do I get this right that you have different kubeconfigs that you want to change to on the fly? So you have a sys variable $KUBECONFIG that points to the file that should be used, correct?

**Reply by SomeAkk** [reply]
> oh yes, it does. Thanks! Appreciate your insight. Do I get this right that you have different kubeconfigs that you want to change to on the fly? So you have a sys variable $KUBECONFIG that points to the file that should be used, correct?

Yep - you are right, but not at all. Except that my $KUBECONFIG not like sys variable (for all computer), but local variable just for single tty. So, in that tty i call my tools and when it run it look to that (just local for tty) $KUBECONFIG variable.

In general it is need to easily connect to any k8s clusters from plugin. 
One solution is to juggle with $KUBECONFIG which maybe looks awkward. 
Another solution - place for your imagination.

**Reply by adietish** [reply]
So you'd launch your tool in different ttys in parallel while using different kubeconfigs for each of them? 
I think that we could mimic this only if you'd launch IDEA from a tty too. 
Furthermore there is no standard for this env var. We'd thus maybe provide a setting that the user could set in the preferences. 
Would that fit your needs?

**Reply by jeffmaury** [reply]
What about the KUBECONFIG env variable ?

**Reply by adietish** [reply]
@jeffmaury The kubernetes client supports the env var "KUBECONFIG" (https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/Config.java#L120). One should be able to launch different IDEA instances with different values for this. On linux/window by launching IDEA directly or on mac via the "open" command. (see https://www.jetbrains.com/help/idea/working-with-the-ide-features-from-command-line.html)

**Reply by SomeAkk** [reply]
> So you'd launch your tool in different ttys in parallel while using different kubeconfigs for each of them?

No, just use tty inside/outside IDEA

> I think that we could mimic this only if you'd launch IDEA from a tty too.

I thing it is very rare case, when anybody start IDEA from tty (just from my view)

> Furthermore there is no standard for this env var. We'd thus maybe provide a setting that the user could set in the preferences.
> Would that fit your needs?

I thing - yes, if at end peoople can see in plugin toolbar another k8s clusters (not only "docker-desktop" or one another) it will be good.

Totaly idea witch KUBECONFIG just idea to thing about it - no more. This is my "user story" and now my workaround which can be used to connect to cluster.

**Reply by adietish** [reply]
ok, thanks for the feedback. I'll stick to #586 and make sure we respect the kubeconfig the env var points to (if present).

**Reply by adietish** [reply]
PR awaiting approval: https://github.com/redhat-developer/intellij-kubernetes/pull/588

**Reply by adietish** [reply]
merged, releasing 0.8.1.
